### PR TITLE
Added user and view fields in model and feature

### PR DESCRIPTION
**Pull request by rkisdp**
Added 2 extra fields with complete flow to added the information into the database.

### CODE REVIEW

**Comment by rkisdp** [comment]
Hi @vishalanandl177, please review the changes and let me know if you find any problem.

File: drf_api_logger/admin.py [hunk 1]
```diff
@@ -96,11 +96,11 @@ def added_on_time(self, obj):
         added_on_time.short_description = 'Added on'
 
         list_per_page = 20
-        list_display = ('id', 'api', 'method', 'status_code', 'execution_time', 'added_on_time',)
+        list_display = ('id', 'api', 'user_id', 'method', 'status_code', 'execution_time', 'added_on_time',)
         list_filter = ('added_on', 'status_code', 'method',)
```
Comment (yhs0602) [inline drf_api_logger/admin.py:100]: I think it will be better if we add 'user_id' in list_filter too!

File: drf_api_logger/utils.py [hunk 2]
```diff
@@ -76,3 +77,17 @@ def mask_sensitive_data(data, mask_api_parameters=False):
             data[key] = [mask_sensitive_data(item) for item in data[key]]
 
     return data
+
+def get_view_from_request(request):
+    try:
+        return f"{resolve(request.path_info)._func_path}"
```
Comment (yhs0602) [inline drf_api_logger/utils.py:83]: I think truncating this into 50 chars will be safer!

**Review by yhs0602 — APPROVED** [review]

File: drf_api_logger/admin.py (hunk 1 above)
Comment (rkisdp) [reply drf_api_logger/admin.py:100]: I'm very sorry, I don't think it's a good idea because it will make django admin dashboard look not good as like of user_id can be very long if there are many users, @vishalanandl177 what's your take on this?

File: drf_api_logger/utils.py (hunk 2 above)
Comment (rkisdp) [reply drf_api_logger/utils.py]: Solved this bug, thanks for flagging this out. 

Do you have any other ideas? I would love to discuss and contribute.

**Comment by ken-turgnate** [comment]
@rkisdp any idea when this PR will merge and a new release will be out?

**Comment by rkisdp** [comment]
> @rkisdp any idea when this PR will merge and a new release will be out?

@ken-turgnate Sorry but I'm not the maintainers of this repository, so can't do or tell you anything this.

**Comment by ken-turgnate** [comment]
Apologies @rkisdp . @vishalanandl177 Any idea on when this land and get released? Thanks

**Comment by rkisdp** [comment]
Hi @vishalanandl177, any update on this PR? if there something you want me to change? I want to add more features in this module, please let me know.

**Comment by tsantor** [comment]
I'm interested in having the user too.  This package is great, but it would be important to know WHO made the call if it was an authenticated user or not.

**Comment by ashwanthbalakrishnan5** [comment]
@vishalanandl177 Are you still maintaining the Repo ?

**Comment by vishalanandl177** [comment]
Yes @ashwanthbalakrishnan5

**Comment by vishalanandl177** [comment]
I cannot merge because doing so may violate certain policies, as observed in various enterprise companies.

**Comment by rkisdp** [comment]
@vishalanandl177 I am very sorry. Can you please tell me what policies I am breaking in this commit? I will fix them.

**Comment by vishalanandl177** [comment]
Based on my recent conversations with several enterprise companies, developers should not have access to user information. Allowing developers to use customer data for personal interests poses a risk.

**Comment by stardust85** [comment]
TL; DR: let's make them optional, disabled by default.

Hi fellow users and devs, I have been employed in multiple SP500 enterprises that takes these risks seriously. In such companies segregation of duties is required for example by Sarbane Oxley Act. Segregation of duties means that if you can change code (you are a developer), you cannot read production DB. Only non-production. So it also means, that you cannot read *any* production data. You simply don't have access to it. You can only access non-production anonymized sample data in non-production DB. If they give production data access to regular developers, they miss the point of Segregation of Duties.

And only in special cases if it not possible to fix the bug without accessing those non-production data, temporary access to production data is allowed, with special approval. 

Source: https://www.securends.com/segregation-of-duties/ (a bit commercial, but correct)

So I think storing ID of the user is perfectly OK. 

However, to be on the safe side, it can be an option in settings.py, so one can it enable explicitly only if needed. Would it be the way to go?
